From e23e16c82f1e333392af129de388c8680417c0f5 Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Tue, 18 Mar 2014 09:18:49 -0700
Subject: [PATCH 493/827] media: xilinx: tpg: Use xvip_set_format_size()

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

Use xvip_set_format_size().

Signed-off-by: Hyun Kwon <hyunk@xilinx.com>
Acked-by: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
(cherry picked from commit 30ed42b705e0bc662f7e86c2005fdc01c6d9cefa)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-tpg.c | 11 ++---------
 1 file changed, 2 insertions(+), 9 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-tpg.c b/drivers/media/platform/xilinx/xilinx-tpg.c
index 7dbd5b9..9b4dcc8 100644
--- a/drivers/media/platform/xilinx/xilinx-tpg.c
+++ b/drivers/media/platform/xilinx/xilinx-tpg.c
@@ -22,11 +22,6 @@
 
 #include "xilinx-vip.h"
 
-#define XTPG_MIN_WIDTH				32
-#define XTPG_MAX_WIDTH				7680
-#define XTPG_MIN_HEIGHT				32
-#define XTPG_MAX_HEIGHT				7680
-
 #define XTPG_CTRL_STATUS_SLAVE_ERROR		(1 << 16)
 #define XTPG_CTRL_IRQ_SLAVE_ERROR		(1 << 16)
 
@@ -192,10 +187,8 @@ static int xtpg_set_format(struct v4l2_subdev *subdev,
 	struct v4l2_mbus_framefmt *__format;
 
 	__format = __xtpg_get_pad_format(xtpg, fh, fmt->pad, fmt->which);
-	__format->width = clamp_t(unsigned int, fmt->format.width,
-				  XTPG_MIN_WIDTH, XTPG_MAX_WIDTH);
-	__format->height = clamp_t(unsigned int, fmt->format.height,
-				   XTPG_MIN_HEIGHT, XTPG_MAX_HEIGHT);
+
+	xvip_set_format_size(__format, fmt);
 
 	fmt->format = *__format;
 
-- 
2.9.3

